src/ray/core_worker/transport/dependency_resolver.cc

```diff
@@ -33,7 +34,24 @@ void InlineDependencies(
       if (it != dependencies.end()) {
         RAY_CHECK(it->second);
         auto *mutable_arg = msg.mutable_args(i);
-        if (!it->second->IsInPlasmaError()) {
+        if (it->second->IsInActorError()) {
+          RAY_CHECK(task.IsActorTask());
+          RAY_LOG(INFO) << "Adding IN_ACTOR dep " << id << " to task " << task.TaskId();
+          // Add the IN_ACTOR error as the metadata value.
+          RAY_CHECK(it->second->HasMetadata());
+          const auto &metadata = it->second->GetMetadata();
+          mutable_arg->set_metadata(metadata->Data(), metadata->Size());
+
+          // Add the RayActorObjectMetadata as a separate field for in-actor
+          // objects only.
+          auto dep = msg.add_in_actor_dependencies();
+          dep->set_object_id(id.Binary());
+          RAY_CHECK(it->second->HasData());
+          const auto &data = it->second->GetData();
+          dep->set_metadata(data->Data(), data->Size());
+
+          dispatch_nccl_send_callback(id, task.ActorId());
```

Comment:
`task.ActorId()` is the receiver actor ID.

Step 3: Driver asks the sender to send the in actor object with id `id` to the receiver actor with actor id `task.ActorId()`.